tests/simulation/raft.cpp

```diff
@@ -0,0 +1,314 @@
+// Copyright 2022 Memgraph Ltd.
```

Comment:
I think some unit tests are still useful, even with this simulation test. This simulation tests is good to test correctness, but it cannot test for example:
* timeouts are configured properly
* the messages between raft participants are the same as we expect them (e.g.: we don't know if the leaders are sending only the necessary logs or not broadcasting the whole log every time)

To test those I think it would be good to have some unit tests. I think the simulator is a good place to start mocking and make sure a single participant behaves as we expect it. If you are unfamiliar with GoogleMock, I can help you to get started. Because of the template functions it will be interesting, but I am sure we can figure out something.

Reply:
Timeouts are tested by the simulator, which asserts that time has not surpassed a certain hard limit by the time that the overall system has completed some amount of expected work despite some requests failing and timing out. If timeouts didn't work as expected, the system would keep running beyond this deadline and fail the test.

Minimality can similarly be asserted on the simulator based on a simulation that asserts a certain expected message count does not go up when the drop percent is 0. I see this as reasonable to add once the coordinator PR is merged, since a few things are in-flux until that point. Unit tests would give worse tests for a lot more effort compared to these two simple one-liner asserts.